CMakeLists.txt

```diff
@@ -160,6 +169,11 @@ add_library(VNNLib SHARED
     ${GENERATED_SRCS}
 )
 
+# Define VNNLIB_BUILD_DLL when building the shared library on Windows (MSVC and MinGW)
+if(WIN32)
```

Comment:
The addition of this parameter seems to be necessary for Windows/MinGW builds on x86/i686 architectures (without this the build failed).